rush/rush/src/utilities/InstallManager.ts

```diff
@@ -46,6 +46,18 @@ export enum InstallType {
 }
 
 /**
+ * Used when determining which versions should be implicitly pinned,
+ * this interface records all version specifiers as well as whether
+ * all versions were optional or not.
+ */
+export interface IImplicitVersion {
```

Comment:
good point. I think the below function needs to be private as well

---
In reply to: [114670964](https://github.com/Microsoft/web-build-tools/pull/172#discussion_r114670964) [](ancestors = 114670964)